pkg/kubelet/images/image_manager.go

```diff
@@ -172,13 +173,14 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 // applyDefaultImageTag parses a docker image string, if it doesn't contain any tag or digest,
 // a default tag will be applied.
 func applyDefaultImageTag(image string) (string, error) {
-	named, err := dockerref.ParseNormalizedNamed(image)
+	_, tag, _, err := parsers.ParseImageName(image)
 	if err != nil {
 		return "", fmt.Errorf("couldn't parse image reference %q: %v", image, err)
 	}
-	_, isTagged := named.(dockerref.Tagged)
-	_, isDigested := named.(dockerref.Digested)
-	if !isTagged && !isDigested {
+	if strings.HasSuffix(image, ":latest") {
```

Comment:
a suffix check seems like it would not work with an image spec that had a latest tag and a digest

Reply:
Can you give an example where that would actually happen?  The code seems to say that you can only specify a tag or digest?

Reply:
for example, an admission plugin that resolves tags to digests and appends the digest are common... so `busybox:latest` would get turned into `busybox:latest@sha256:...`

Reply:
thanks for that.  Pushed up a test case for this.